### PR TITLE
Remove EventSourceActivity

### LINKED ISSUE #53177: EventSourceActivity doesn't seem to be used anywhere
The `EventSourceActivity` class doesn't seem to be used by anything in CoreLib. Since we run linker on corelib during build, the class is actually removed from the product and can't be used by tests or product - which sort of proves that it's not needed.
Should we remove it?

### PR DESCRIPTION
**Pull request by hrrrrustic**
close #53177

### CODE REVIEW

**Comment by stephentoub** [comment]
This file is included in both System.Private.CoreLib.Shared.projitems and Microsoft.Diagnostics.Tracing.EventSource.Redist.csproj and will need to be removed from both.  As is, I'd expect the build to fail.

**Comment by hrrrrustic** [comment]
Ah yes, checked only .cs files 😄

File: src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems [hunk 1]
```diff
@@ -96,7 +96,6 @@
     <Compile Include="$(MSBuildThisFileDirectory)System\Boolean.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Buffer.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Buffers\ArrayPool.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)System\Buffers\ArrayPoolEventSource.cs" />
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems:99]: This definitely should not be deleted.

Comment (hrrrrustic) [reply src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems]: Whoops, you didn't see anything 👀

**Review by stephentoub — APPROVED** [review]
Thanks.

**Pull request merged**